capstone/src/main/java/com/google/sps/search/SearchPredictor.java

```diff
@@ -116,7 +147,52 @@ public int compare(Map.Entry<String, Integer> a, Map.Entry<String, Integer> b) {
       partialName = partialName.toUpperCase();
       if (partialName.equals(firstAndLastName[0].toUpperCase())
           || partialName.equals(firstAndLastName[1].toUpperCase())) {
-        int score = namesScore.get(name) + COMPLETE_PARTIAL_NAME_MATCH;
+        double score = namesScore.get(name) + COMPLETE_PARTIAL_NAME_MATCH;
+        namesScore.put(name, score);
+      }
+    }
+    return namesScore;
+  }
+
+  /**
+   * Adds names in a set to the map alongside its score.
+   *
+   * @param namesScore map from name to score
+   * @param set set of suggested names
+   * @param increment score
+   * @return map from name to score
+   */
+  private Map<String, Double> addToMap(
+      Map<String, Double> namesScore, Set<String> set, double increment) {
+    for (String name : set) {
+      if (!namesScore.containsKey(name)) {
+        namesScore.put(name, increment);
+      } else {
+        namesScore.put(name, namesScore.get(name) + 1);
+      }
+    }
+    return namesScore;
+  }
+
+  /**
+   * Adds names from the name to led map to the name to score map.
+   *
+   * @param namesScore map from name to score
+   * @param nameToLedMap map from name to its led distance from input
+   * @return map from name to score
+   */
+  private Map<String, Double> addToMap(
+      Map<String, Double> namesScore, Map<String, Integer> nameToLedMap) {
+    for (String name : nameToLedMap.keySet()) {
+      double score = 0;
+      /* Get the reciprocal of the edit distance so that names with greater edit distance have a
+      lower score */
+      if (nameToLedMap.get(name) != 0) {
+        score = 1 / nameToLedMap.get(name);
+      }
+      if (namesScore.containsKey(name)) {
+        namesScore.put(name, namesScore.get(name) + score);
+      } else {
         namesScore.put(name, score);
       }
     }
```

Comment:
This is clever and I like it (addToMap + the variations) However, the absense of the `double increment` parameter now triggers separate reciprocal logic, so I would be careful overloading the same method name `addToMap` since the behavior is actually pretty different. I would instead call this something like `addToMapWithEditDistance` or something that clearly indicates that this logic is unique.